### PR TITLE
enforce dtype on quantity

### PR DESCRIPTION
**Pull request by ipelupessy**
I had this lying around..it seems sensible, but has a performance penalty

### CODE REVIEW

**Comment by rieder** [comment]
Definitely seems sensible

**Comment by rieder** [comment]
Codefactor was complaining about some issues that weren't changed in this PR. Not sure why, but I fixed most of these. 
I'm happy for this to be merged. I doubt the performance penalty would be large and I think it's better to enforce the dtypes.

File: src/amuse/units/quantities.py [hunk 1]
```diff
@@ -10,8 +10,7 @@
 from amuse.support.core import compare_version_strings
 from amuse.units import core
 from amuse.units.si import none
-from amuse.units.core import zero_unit
-
+from amuse.units.core import zero_unit:
```
Comment (ipelupessy) [inline src/amuse/units/quantities.py:13]: colon?

**Comment by ipelupessy** [comment]
the dtype thing eventually has to be reworked ..e.g. right now its not conisstent what the result of the operations with dtypes is...also I think maybe dtype should move to quantity...(in any case, all this is beyond the scope of this PR)

File: src/amuse/units/quantities.py (hunk 1 above)
Comment (rieder) [reply src/amuse/units/quantities.py:13]: typo I think. will remove.

**Pull request merged**